project/ValidatePullRequest.scala

```diff
@@ -3,19 +3,171 @@
+import org.kohsuke.github._
+import sbt.Keys._
+import sbt._
+
+import scala.collection.immutable
+import scala.util.matching.Regex
 
 object ValidatePullRequest extends AutoPlugin {
 
+  val ValidatePR = config("pr-validation") extend Test
+
+  /*
+    Assumptions:
+      Env variables set "by Jenkins" are assumed to come from this plugin:
+      https://wiki.jenkins-ci.org/display/JENKINS/GitHub+pull+request+builder+plugin
+   */
+
+  // settings
+  val PullIdEnvVarName = "ghprbPullId" // Set by "GitHub pull request builder plugin"
+
+  val TargetBranchEnvVarName = "PR_TARGET_BRANCH"
+  val TargetBranchJenkinsEnvVarName = "ghprbTargetBranch"
+  val sourceBranch = settingKey[String]("Branch containing the changes of this PR")
+
+  val SourceBranchEnvVarName = "PR_SOURCE_BRANCH"
+  val SourceBranchJenkinsEnvVarName = "ghprbSourceBranch"
+  val targetBranch = settingKey[String]("Branch with which the PR changes should be diffed against")
+
+  // asking github comments if this PR should be PLS BUILD ALL
+  val githubEnforcedBuildAll = taskKey[Boolean]("Checks via GitHub API if comments included the PLS BUILD ALL keyword")
+  val buildAllKeyword = taskKey[Regex]("Magic phrase to be used to trigger building of the entire project instead of analysing dependencies")
+
+  // determining touched dirs and projects
+  val changedDirectories = taskKey[immutable.Set[String]]("List of touched modules in this PR branch")
+  val projectIsAffectedByChanges = taskKey[Boolean]("True if this project is affected by the PR and should be rebuilt")
+
+  // running validation
   val validatePullRequest = taskKey[Unit]("Additional tasks for pull request validation")
 
   override def trigger = allRequirements
 
   override lazy val projectSettings = Seq(
-    validatePullRequest := (),
-    validatePullRequest <<= validatePullRequest.dependsOn(test in Test),
+    testOptions in ValidatePR += Tests.Argument("-l akka.testkit.PerformanceTest"),
+    testOptions in ValidatePR += Tests.Argument("-l akka.testkit.LongRunningTest"),
+    testOptions in ValidatePR += Tests.Argument("-l akka.testkit.TimingTest"),
+
+    targetBranch in ValidatePR := {
+        sys.env.get(TargetBranchEnvVarName) orElse
+        sys.env.get(TargetBranchJenkinsEnvVarName) getOrElse // Set by "GitHub pull request builder plugin"
+        "master"
+    },
+
+    sourceBranch in ValidatePR := {
+      sys.env.get(SourceBranchEnvVarName) orElse
+      sys.env.get(SourceBranchJenkinsEnvVarName) getOrElse // Set by "GitHub pull request builder plugin"
+      "HEAD"
+    },
+
+    changedDirectories in ValidatePR := {
+      val log = streams.value.log
+
+      val targetId = (targetBranch in ValidatePR).value
+      val prId = (sourceBranch in ValidatePR).value
+
+      // TODO could use jgit
+      log.info(s"Comparing [$targetId] with [$prId] to determine changed modules in PR...")
+      val gitOutput = "git diff %s..%s --stat".format(targetId, prId).!!.split("\n")
```

Comment:
This produces the following output on my machine when converted to List:
```
List( .../src/test/scala/akka/camel/internal/ActivationTrackerTest.scala     | 3 ++-,  1 file changed, 2 insertions(+), 1 deletion(-))
```

which then of course makes the code below match nothing. I propose to change the command to `git diff %s..%s --name-only` which outputs
```
List(akka-camel/src/test/scala/akka/camel/internal/ActivationTrackerTest.scala)
```

allowing to simplify the code below.